lib/creator/index.js

```diff
@@ -1,42 +1,36 @@
-const validateSchema = require('./utils/validateSchema.js');
-const webpackOptionsSchema = require('./utils/webpackOptionsSchema.json');
-const WebpackOptionsValidationError = require('./utils/WebpackOptionsValidationError');
-const initTransform = require('./init-transform');
-const chalk = require('chalk');
+const yeoman = require('yeoman-environment');
+const Generator = require('yeoman-generator');
+const defaultGenerator = require('./yeoman/webpack-generator');
+const WebpackAdapter = require('./yeoman/webpack-adapter');
+const runTransform = require('./transformations/index');
+
 /*
 * @function creator
 *
-* Main function to build up a webpack configuration.
-* Either throws an error if it doesn't match the webpack schema,
-* or validates the filepaths of the options given.
-* If a package is supplied, it finds the path of the package and runs inquirer
+* Runs yeoman and runs the transformations based on the object
+* built up from an author/user
 *
-* @param { Array } pkgPaths - An Array of packages to run
-* @param { <object|null> } opts - An object containing webpackOptions or nothing
-* @returns { <Function|Error> } initTransform - Initializes the scaffold in yeoman
+* @param { String } options - An path to the given generator
+* @returns { Function } runTransform - Run transformations based on yeoman prompt
 */
 
-module.exports = function creator(pkgPaths, opts) {
-	// null, config -> without package
-	// addon, null -> with package
-	// we're dealing with init, we need to change this later, as it may have been emptied by yeoman
-	if(!pkgPaths && !opts) {
-		initTransform();
-	}
-	else if(pkgPaths) {
-		// this example app actually needs a refactor in order for it to work
-		initTransform(pkgPaths);
-	}
-	else if(!pkgPaths && opts) {
-		console.log(opts);
-		// scaffold is done
-		/*
-		const webpackOptionsValidationErrors = validateSchema(webpackOptionsSchema, initialWebpackConfig);
-		if (webpackOptionsValidationErrors.length) {
-			throw new WebpackOptionsValidationError(webpackOptionsValidationErrors);
-		} else {
-			process.stdout.write('\n' + chalk.green('Congratulations! Your new webpack config file is created!') + '\n');
-		}
-		*/
+module.exports = function creator(options) {
+	const env = yeoman.createEnv(null, null, new WebpackAdapter());
+
+	if(options) {
+		env.register(require.resolve(options), 'npm:app');
+
+		env.run('npm:app')
+		.on('end', () => {
```

Comment:
This is an arrow function, so you could either just one-line it and leave out the return, or since it's an event, you probably don't need the return at all.